packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1515,3 +1550,52 @@ export enum BuildEnvironmentVariableType {
    */
   SECRETS_MANAGER = 'SECRETS_MANAGER'
 }
+
+export interface FileSystemConfig {
+  readonly location: CfnProject.ProjectFileSystemLocationProperty;
+}
+
+/**
+ * The abstract interface of a CodeBuild FileSystemLocation.
+ */
+export interface IFileSystemLocation {
+  bind(scope: Construct, project: IProject): FileSystemConfig;
+}
+
+/**
+ * FileSystemLocation provider definition for a CodeBuild Project.
+ */
+export class FileSystemLocation {
+  public static efs(props: IGenericFileSystemLocationProps): IFileSystemLocation {
+    return new EfsFileSystemLocation(props);
+  }
+}
+
+/**
+ * EfsFileSystemLocation definition for a CodeBuild project.
+ */
+class EfsFileSystemLocation implements IFileSystemLocation {
+  constructor(private readonly props: IGenericFileSystemLocationProps) {}
+
+  public bind(_scope: Construct, _project: IProject): FileSystemConfig {
+    return {
+      location: {
+        identifier: this.props.identifier,
+        location: this.props.location,
+        mountOptions: this.props.mountOptions,
+        mountPoint: this.props.mountPoint,
+        type: 'EFS',
+      },
+    };
+  }
+}
+/**
+ * The abstract interface of a CodeBuild FileSystemLocation.
+ */
+export interface IGenericFileSystemLocationProps {
+  readonly identifier: string;
+  readonly location: string;
+  readonly mountOptions?: string;
+  readonly mountPoint: string;
+  readonly type: string;
```

Comment:
`type` should not be here. It's always `EFS` in this case (the property is also never used).